xds/internal/resolver/serviceconfig.go

```diff
@@ -93,12 +95,13 @@ func serviceConfigJSON(activeClusters map[string]*clusterInfo) (string, error) {
 }
 
 type route struct {
-	action wrr.WRR
-	m      *compositeMatcher // converted from route matchers
+	m                 *compositeMatcher // converted from route matchers
+	clusters          wrr.WRR
+	maxStreamDuration time.Duration
 }
 
 func (r route) String() string {
-	return r.m.String() + "->" + fmt.Sprint(r.action)
+	return fmt.Sprintf("%s -> { clusters: %v, maxStreamDuration: %v }", r.m.String(), r.clusters, r.maxStreamDuration)
```

Comment:
Add the `String` method to the `wrr.WRR` interface.

Actually, why does this work since `String` is not part of the interface?

Reply:
The interface is `fmt.Stringer`.  I don't think we need to explicitly add `Stringer` everywhere we want to print things; it's a magic part of standard library stuff.

Reply:
Yeah, sorry. I thought you were calling `String()` on `clusters`.